common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -70,117 +70,70 @@ public static String modifyMultipleMatchKeyword(String query){
     public static final String STANDARD_CONCEPT_CODE = "S";
     public static final String CLASSIFICATION_CONCEPT_CODE = "C";
 
-    public Slice<Concept> searchConcepts(String query,
-                                         StandardConceptFilter standardConceptFilter, List<String> vocabularyIds,
-                                         List<String> domainIds, int limit, int minCount) {
+    public Slice<Concept> searchConcepts(String query, StandardConceptFilter standardConceptFilter, List<String> vocabularyIds, List<String> domainIds, int limit, int minCount, List<Long> matchedConceptIds) {
```

Comment:
You're not actually using query in here anymore. If you're going to use this in clause approach, you should require matchedConceptIds to be set and remove the query parameter. (Though see above about why that's probably not a good idea.)